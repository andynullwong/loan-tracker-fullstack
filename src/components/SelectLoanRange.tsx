import { DatePicker, Space } from "antd";

const { RangePicker } = DatePicker;

const SelectLoanRange = () => {
  return (
    <Space direction="vertical" size={12}>
      <RangePicker onCalendarChange={(e) => console.log(e)} />
    </Space>
  );
};

export default SelectLoanRange;
