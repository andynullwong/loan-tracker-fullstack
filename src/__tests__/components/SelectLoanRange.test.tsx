import React from "react";
import { act, render } from "@testing-library/react";
import SelectLoanRange from "../../components/SelectLoanRange";

test("renders SelectLoanRange correctly", () => {
  const setFromDate = jest.fn();
  const setToDate = jest.fn();

  let component = null;

  act(() => {
    component = render(
      <SelectLoanRange setFromDate={setFromDate} setToDate={setToDate} />
    );
  });

  expect(component).toMatchSnapshot();
});
