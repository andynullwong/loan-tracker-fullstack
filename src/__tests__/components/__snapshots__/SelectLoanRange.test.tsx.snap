// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SelectLoanRange correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ant-space css-dev-only-do-not-override-1vr7spz ant-space-vertical"
        style="column-gap: 12px; row-gap: 12px;"
      >
        <div
          class="ant-space-item"
        >
          <div
            class="ant-picker ant-picker-range css-dev-only-do-not-override-1vr7spz"
          >
            <div
              class="ant-picker-input ant-picker-input-active"
            >
              <input
                autocomplete="off"
                placeholder="Start month"
                readonly=""
                size="12"
                value=""
              />
            </div>
            <div
              class="ant-picker-range-separator"
            >
              <span
                aria-label="to"
                class="ant-picker-separator"
              >
                <span
                  aria-label="swap-right"
                  class="anticon anticon-swap-right"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="swap-right"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="0 0 1024 1024"
                    width="1em"
                  >
                    <path
                      d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"
                    />
                  </svg>
                </span>
              </span>
            </div>
            <div
              class="ant-picker-input"
            >
              <input
                autocomplete="off"
                placeholder="End month"
                readonly=""
                size="12"
                value=""
              />
            </div>
            <div
              class="ant-picker-active-bar"
              style="left: 0px; width: 0px; position: absolute;"
            />
            <span
              class="ant-picker-suffix"
            >
              <span
                aria-label="calendar"
                class="anticon anticon-calendar"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="calendar"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ant-space css-dev-only-do-not-override-1vr7spz ant-space-vertical"
      style="column-gap: 12px; row-gap: 12px;"
    >
      <div
        class="ant-space-item"
      >
        <div
          class="ant-picker ant-picker-range css-dev-only-do-not-override-1vr7spz"
        >
          <div
            class="ant-picker-input ant-picker-input-active"
          >
            <input
              autocomplete="off"
              placeholder="Start month"
              readonly=""
              size="12"
              value=""
            />
          </div>
          <div
            class="ant-picker-range-separator"
          >
            <span
              aria-label="to"
              class="ant-picker-separator"
            >
              <span
                aria-label="swap-right"
                class="anticon anticon-swap-right"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="swap-right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 1024 1024"
                  width="1em"
                >
                  <path
                    d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            class="ant-picker-input"
          >
            <input
              autocomplete="off"
              placeholder="End month"
              readonly=""
              size="12"
              value=""
            />
          </div>
          <div
            class="ant-picker-active-bar"
            style="left: 0px; width: 0px; position: absolute;"
          />
          <span
            class="ant-picker-suffix"
          >
            <span
              aria-label="calendar"
              class="anticon anticon-calendar"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="calendar"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                />
              </svg>
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
